parser-java/api/src/main/java/org/jboss/forge/addon/parser/java/ui/AbstractJavaSourceCommand.java

```diff
@@ -63,13 +59,20 @@
          @FacetConstraint(value = JavaSourceFacet.class, type = FacetConstraintType.REQUIRED)
 })
 public abstract class AbstractJavaSourceCommand<SOURCETYPE extends JavaSource<?>> extends AbstractProjectCommand
+      implements JavaSourceDecorator<SOURCETYPE>
 {
    private UIInput<String> targetPackage;
    private UIInput<String> named;
    private UIInput<Boolean> overwrite;
    private UIInput<String> extendsType;
    private UIInputMany<String> implementsType;
 
+   private JavaSourceDecorator<SOURCETYPE> delegate = this;
+
+   public void setJavaSourceDecorator(JavaSourceDecorator<SOURCETYPE> delegate) {
```

Comment:
What if you want your new decorator implementation to keep the original behavior? Perhaps pass "this" as a parameter to the JavaSourceDecorator functional method?